lib/ansible/module_utils/rabbitmq.py

```diff
@@ -17,3 +35,201 @@ def rabbitmq_argument_spec():
+
+        if self.host is not None:
+            self.build_url()
+
+        self.conn_channel = self.connect_to_rabbitmq()
+
+    def check_required_library(self):
+        if not HAS_PIKA:
+            self.module.fail_json(msg="Unable to find 'pika' Python library which is required.")
+
+    def check_host_params(self):
+        # Fail if url is specified and other conflicting parameters have been specified
+        if self.params['url'] is not None and any(self.params[k] is not None for k in ['proto', 'host', 'port', 'password', 'username', 'vhost']):
+            self.module.fail_json(msg="url and proto, host, port, vhost, username or password cannot be specified at the same time.")
+
+        # Fail if url not specified and there is a missing parameter to build the url
+        if self.params['url'] is None and any(self.params[k] is None for k in ['proto', 'host', 'port', 'password', 'username', 'vhost']):
+            self.module.fail_json(msg="Connection parameters must be passed via url, or,  proto, host, port, vhost, username or password.")
+
+    @staticmethod
+    def rabbitmq_argument_spec():
+        return dict(
+            url=dict(default=None, type='str'),
+            proto=dict(default=None, type='str'),
+            host=dict(default=None, type='str'),
+            port=dict(default=None, type='int'),
+            username=dict(default=None, type='str'),
+            password=dict(default=None, type='str', no_log=True),
+            vhost=dict(default=None, type='str'),
+            queue=dict(default=None, type='str')
+        )
+
+    ''' Consider some file size limits here '''
+    def _read_file(self, path):
+        try:
+            fh = open(path, "rb").read()
+        except IOError:
+            self.module.fail_json(msg="Unable to open file %s." % path)
+
+        return fh
+
+    @staticmethod
+    def _check_file_mime_type(path):
+        mime = MimeTypes()
+        return mime.guess_type(path)
+
+    def build_url(self):
+        self.url = '{0}://{1}:{2}@{3}:{4}/{5}'.format(self.proto,
+                                                      self.username,
+                                                      self.password,
+                                                      self.host,
+                                                      self.port,
+                                                      self.vhost)
+
+    def connect_to_rabbitmq(self):
+        """
+        Function to connect to rabbitmq using username and password
+        """
+        try:
+            parameters = pika.URLParameters(self.url)
+        except Exception as e:
+            self.module.fail_json(msg="URL malformed: %s" % to_native(e))
+
+        try:
+            self.connection = pika.BlockingConnection(parameters)
+        except Exception as e:
+            self.module.fail_json(msg="Connection issue: %s" % to_native(e))
+
+        try:
+            conn_channel = self.connection.channel()
```

Comment:
Should not that be `self.conn_channel = self.connection.channel()` (then remove the `return` statement and assignment at the end of `__init__`).

Reply:
Updated.  Thanks!